src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -165,6 +166,21 @@ EmberAfStatus OnOffServer::setOnOffValue(chip::EndpointId endpoint, uint8_t comm
         {
             emberAfOnOffClusterLevelControlEffectCallback(endpoint, newValue);
         }
+#endif
+#ifdef EMBER_AF_PLUGIN_MODE_SELECT
+        // If OnMode is not a null value, then change the current mode to it.
+        ModeSelect::Attributes::OnMode::TypeInfo::Type onMode;
+        status = ModeSelect::Attributes::OnMode::Get(endpoint, onMode);
+        if (status != EMBER_ZCL_STATUS_SUCCESS)
+        {
+            emberAfOnOffClusterPrintln("ERR: reading onMode %x", status);
+            return status;
+        }
+        if (!onMode.IsNull() && onMode.HasValidValue())
```

Comment:
Why is this checking HasValidValue?  That will never be false here....

Reply:
```suggestion
        if (!onMode.IsNull())
```